### DISCUSSION TITLE
Error updating 1.7.9 to 1.8

### DISCUSSION THREAD

**Discussion by diplatis** [YOURLS/YOURLS#2857] [Q&A] [answered]
When I tried updating from 1.7.9 to 1.8 I got the following error after prompting for database update:
`SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes`

After reloading the page, pressing F5, the update completed successfully without getting an error.
Is everything ok? I am concerned if the database update had been successful or there are changes pending.
Is there any way to check this?
YOURLS run on PHP 7.3 and MariaDB 10.1.

**Comment by dgw** [comment, marked as answer]
There are numerous issues about this particular MySQL error. #2115 should give you a good sense of how to prevent it from happening in the future. @ozh is there a clean way to re-run the database upgrade steps?

**Reply by ozh** [reply]
I'd say you can either manually run [the queries](https://github.com/YOURLS/YOURLS/blob/1.8/includes/functions-upgrade.php#L100-L107) (using something like phpmyadmin for example), or simply compare the current table structures with what they should be when [created from scratch](https://github.com/YOURLS/YOURLS/blob/1.8/includes/functions-install.php#L212-L246)

This being said, I'm a bit surprised this bug still happens, I thought it was fixed since the decrease of keyword to `varchar(100)` :)

**Reply by diplatis** [reply]
I runned the queries, all of them returned empty set of changes.
Everything seems ok.

**Comment by GerZah** [comment]
Unearthing this old thread – however, to whom it concerns: Same issue – desired upgrade from 1.7.9 to 1.8 on PHP 7.4, MariaDB 10.11 (not 10.1), but seeing the same error message.

Solution for me: Upgrade straight from 1.7.9 to 1.8.2: In that case, the database upgrade went through for me.
